src/acvp_ml_kem.c

```diff
@@ -190,18 +196,35 @@ static ACVP_RESULT acvp_ml_kem_init_tc(ACVP_CTX *ctx,
                 ACVP_LOG_ERR("Hex conversion failure (m)");
                 return rv;
             }
-        } else {
-            rv = acvp_hexstr_to_bin(dk, stc->dk, ACVP_ML_KEM_TMP_BYTE_MAX, &(stc->dk_len));
+            /* fallthru */
+        case ACVP_ML_KEM_FUNCTION_ENC_KEYCHECK:
+            rv = acvp_hexstr_to_bin(ek, stc->ek, ACVP_ML_KEM_TMP_BYTE_MAX, &(stc->ek_len));
             if (rv != ACVP_SUCCESS) {
-                ACVP_LOG_ERR("Hex conversion failure (dk)");
+                ACVP_LOG_ERR("Hex conversion failure (ek)");
                 return rv;
             }
-
+            break;
+        case ACVP_ML_KEM_FUNCTION_DECAPSULATE:
+            stc->c = calloc(ACVP_ML_KEM_TMP_BYTE_MAX, sizeof(unsigned char));
```

Comment:
This creates a double calloc of stc->c so the original calloc at line 179 is lost